modules/tensor_mechanics/src/actions/DomainIntegralAction.C

```diff
@@ -68,6 +71,13 @@ validParams<DomainIntegralAction>()
       false,
       "Calculate an equivalent K from KI, KII and KIII, assuming self-similar crack growth.");
   params.addParam<bool>("output_q", true, "Output q");
+  params.addParam<bool>("solid_mechanics",
+                        false,
+                        "Set to true if the solid_mechanics system is "
+                        "used. This option is only needed for "
+                        "interaction integrals.");
```

Comment:
I'm just about ready to merge this, but I'm concerned about breaking existing solid_mechanics models, which will need to add this. I wonder -- is there any way we can detect that a model is using solid_mechanics so that the user doesn't need to specify this?  I assume those models will fail if this isn't added, but probably in a way that does not provide very good feedback to the user about what the problem was.

Reply:
I am not sure how to detect whether the input file uses solid or tensormechanics objects. I can check to see if there is any method to do that. 

If solid_mechanics option is not set to true, then it gives an error saying material property stress does not exist. I agree that error is not very descriptive.

Reply:
In the constructor I can check to see if the stress/strain material property is of type SymmTensor or RankTwoTensor and set the solid_mechanics flag accordingly. Though I am not sure whether the material properties are declared before the action is called. 

If the action thing doesn't work, I can do a similar check in the InteractionIntegral.C (tensor_mech version) to produce a better error if the material property stress is not found.

Reply:
The Action gets run before anything else, so you're probably out of luck trying to detect that within the Action. You should be able to do that check in the InteractionIntegral postprocessor, though.

Reply:
I have added a check in InteractionIntegral.C to give a more descriptive error if stress material property in RankTwoTensor form is not found. I have also added a test to the interaction_integral tests in solid_mechanics so make sure the error works as expected.